repo/fsrepo/migrations/fetch.go

```diff
@@ -0,0 +1,272 @@
+package migrations
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"runtime"
+	"strings"
+)
+
+const (
+	// Current dirstibution to fetch migrations from
+	CurrentIpfsDist = "/ipfs/Qme8pJhBidEUXRdpcWLGR2fkG5kdwVnaMh3kabjfP8zz7Y"
+
+	envIpfsDistPath = "IPFS_DIST_PATH"
+
+	// Distribution
+	gatewayURL = "https://ipfs.io"
+	ipfsDist   = "/ipns/dist.ipfs.io"
+
+	// Maximum download size
+	fetchSizeLimit = 1024 * 1024 * 512
+)
+
+type limitReadCloser struct {
+	io.Reader
+	io.Closer
+}
+
+var ipfsDistPath string
+
+func init() {
+	SetIpfsDistPath("")
+}
+
+// SetIpfsDistPath sets the ipfs path to the distribution site.  If an empty
+// string is given, then the path is set using the IPFS_DIST_PATH environ
+// veriable, or the default dns link value if that is not defined.
+func SetIpfsDistPath(distPath string) {
```

Comment:
It was intended to simplify APIs.  Multiple APIs need to know the location of the distribution site, this allows the location to be set once, if needed, instead of passing another value into multiple APIs.  I also assume this was why it was previously a global variable, and not a function argument: https://github.com/ipfs/go-ipfs/blob/master/repo/fsrepo/migrations/migrations.go#L18

Would you prefer to see the distribution location passed into all the APIs that need it?

Reply:
This is now passed to the fetcher(s) and is not a global.